tests/ci/functional_test_check.py

```diff
@@ -111,17 +112,21 @@ def get_run_command(
 
 
 def get_tests_to_run(pr_info):
-    result = set([])
+    result = set()
 
     if pr_info.changed_files is None:
         return []
 
     for fpath in pr_info.changed_files:
-        if "tests/queries/0_stateless/0" in fpath:
-            logging.info("File %s changed and seems like stateless test", fpath)
+        if re.match(r"tests/queries/0_s[a-z]+/[0-9]{5}[a-zA-Z_]+\.reference$", fpath):
+            logging.info(f"File '{fpath}' is changed and seems like a test")
             fname = fpath.split("/")[3]
             fname_without_ext = os.path.splitext(fname)[0]
+            # Add '.' to the end of the test name not to run all tests with given prefix.
+            # We do not want to specify exact file extension not to complicate matching and look only for .reference
```

Comment:
What if the `.reference` file is not changed, but the query does?

Reply:
Indeed, thanks. I returned logic back, just with the change `tests/queries/0_stateless/0` to `tests/queries/0_stateless/[0-9]{5}`